doc/incus-cli: Add CLI configuration file reference (#3483)
lxc/incus: doc/cli_configs.md

> (client-configs)=

Reviewer: I’d put a singular (same with the file name)

# CLI configuration file
The Incus CLI client uses a configuration file to store environment data and modify it's behavior. This documentation describes the file structure and it's options.

## The configuration file

> By default, the Incus CLI will use the `HOME/.config/incus/config.yml`, but it can also use an arbitrary file provided by the `INCUS_CONF` environment variable.

Reviewer: Maybe `$HOME` and not `HOME`?

> If the file or it's path don't exist, it will use the default:

Reviewer: ```suggestion
If the file or its path don't exist, it will use the default:
```

```yaml
default-remote: local
remotes:
  images:
    protocol: simplestreams
    public: true
    addr: https://images.linuxcontainers.org
  local:
    protocol: incus
    public: false
    addr: unix://
aliases: {}
defaults:
  list_format: ""
  console_type: ""
  console_spice_command: ""
  no_color: false
```

## `remotes`

> Incus remote information managed by [`incus remote`](incus_remote.md) commands is stored in the remote section. Remotes may have the following configurations.

Reviewer: ```suggestion
Incus remote information managed by [`incus remote`](incus_remote.md) commands is stored in the `remotes` section. Remotes may have the following configuration keys.
```

### `addr`

> Address of the remote server. Accepts IPs, FQDN and URLs. Multiple addresses can be provided, separated by commas `addr1,addr2,...`.

Reviewer: ```suggestion
Address of the remote server. Accepts IPs, FQDNs and URLs. Multiple addresses can be provided, separated with commas `addr1,addr2,...`.
```

### `last_working_address`

> Defines the first address the CLI will try using when a remote has multiple addresses. It's used by the CLI to give preference to an address that has more probability of success.

Reviewer: It’s not completely about probabilities, it’s rather the last address the CLI tried that worked. I feel like you’re describing an observable effect of what it does, rather than what it actually does.

Author: I see, I was trying to describe the purpose of this configuration by describing one of it's advantages. What do you think about "It's used by the CLI to give preference to an address that has already worked", or remove this phrase at all?

Reviewer: The purpose is just that “it worked before, so we should try to use it now”. Honestly, the nature of this key is in its title, so I wouldn’t spend too much time on finding the perfect phrasing :)

### `auth_type`
Method that the client will use to authenticate to the server API. Accepts `tls` and `oidc`.

### `keepalive`
Timeout in seconds that the `keepalive` feature will maintain the connection opened.

### `project`

> Default project used by the CLI in that remote. It's defined by the [`incus project switch`](incus_project_switch.md) command. It's overwritten by the `--project` flag and the `INCUS_PROJECT` environment variable.

Reviewer: ```suggestion
Default project used by the CLI when using that remote. It's defined by the [`incus project switch`](incus_project_switch.md) command. It's overwritten by the `--project` flag and the `INCUS_PROJECT` environment variable.
```

### `protocol`
The protocol the CLI will use to communicate with the remote. This defines if the remote will be used as an Incus or image server. Accepts the following options:

> - `incus`: regular Incus server accessible over the network to provide remote access or images

Reviewer: ```suggestion
- `incus`: regular Incus server accessible over the network to provide access to instances or images
```

Author: In this protocol type the incus remote can manage other resources than instances, like volumes, cluster members etc., is there a way of describing all of those actions? Like "access to the Incus server", or "access to the Incus cluster".

Reviewer: I’d say we don’t want to get too boring. “regular private Incus server” could also work; in the end, we just want something users can quickly understand.

- `oci`: [Open Container Initiative (OCI)](https://opencontainers.org/) server that provides application container images

- `public`: Incus server accessible over the network to provide images only

- `simplestream`: [Simple Stream](https://git.launchpad.net/simplestreams/tree/) server that provide images

### `credentials_helper`

> Defines an helper command that will handle OCI service authentication.

Reviewer: ```suggestion
Defines a helper command that will handle OCI service authentication.
```

### `public`

> Defines if the remote is an image only server (public).

Reviewer: ```suggestion
Defines if the remote is an image-only server (public).
```

## `default_remotes`
Defines the default Incus remote used by the CLI to execute commands. It's defined by the [`incus remote switch`](incus_remote_switch.md) command. This configuration is overwritten by the environment variable `INCUS_REMOTE` or by the `<remote>:` prefix.

## `aliases`

> Stores the alias managed by the [`incus alias`](incus_alias.md) command. Each alias is a key value pair that consists of the alias name and the alias command (enclosed by quotes).

Reviewer: ```suggestion
Stores the alias managed by the [`incus alias`](incus_alias.md) command. Each alias is a key/value pair that consists of the alias name and the alias command (enclosed by quotes).
```

## `defaults`

> Determine the default CLI behavior for a certain CLI commands.

Reviewer: ```suggestion
Determines the default CLI behavior for certain CLI commands.
```

Your other descriptions used the 3rd person
